examples/complete/main.tf

```diff
@@ -10,6 +10,35 @@ module "resource_group" {
   existing_resource_group_name = var.resource_group
 }
 
+##############################################################################
+# Secrets Manager Certificate Setup
+##############################################################################
+
+module "sm_crn" {
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.1.0"
+  crn     = var.existing_secrets_manager_crn
+}
+
+module "secrets_manager_cert" {
+  source                            = "terraform-ibm-modules/secrets-manager-public-cert/ibm"
+  version                           = "1.2.1"
+  secrets_manager_guid              = module.sm_crn.service_instance
+  secrets_manager_region            = module.sm_crn.region
+  cert_name                         = "${var.prefix}-kmip-cert"
+  secrets_manager_dns_provider_name = var.secrets_manager_dns_provider_name
+  secrets_manager_ca_name           = var.secrets_manager_ca_name
+  cert_common_name                  = "${var.prefix}.${var.domain_name}"
+}
+
+data "ibm_sm_public_certificate" "kmip_cert" {
+  depends_on        = [module.secrets_manager_cert]
```

Comment:
Always leave a code comment explaining why you used `depends_on` since its a bad practise and can lead to issues with dependency tree. Does our `secrets-manager-public-cert` module not have the outputs you need so you can avoid doing a data lookup? If it doesn't, then its a gap and should be plugged

Reply:
the certificate modules do not include any outputs that provide the certificate details, just references to IDs to access the certificates via data lookups. I have modified this data lookup to use the secret ID so we no longer need an explicit depends on here, but one is still required between the cert engine module and cert module because the cert engine has no outputs. I've added a comment in the code explaining the need for the explicit depends on there.